### PR TITLE
fix(wallet): several regressions due to recent store refactorings

### LINKED ISSUE #16089: [WalletConnect] Incorrect authentication method icon on sign modal
# Bug Report

## Description

The authentication icon on the `Sign` button is not correct when using a local profile and importing a keycard wallet account.

## Steps to reproduce

1. Import a wallet account from keycard
2. Connect a dApp to this account
3. Request any sign method
4. Go to sign request dialog in Status

#### Expected behavior

The authentication icon on the `Sign` button shows the keycard icon

#### Actual behavior

The authentication icon on the `Sign` button shows the password (or any profile authentication method)

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

(best review commit by commit, also see my inline code comments below):
- TransactionDelegate: fix `transactionType` function
- AddEditSavedAddressPopup: fix validating account
- DappSignRequestModal: fix passing `loginType` from the correct RootStore + an evil typo fix

Fixes #16089

### Affected areas

Wallet(Connect)

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Architecture guidelines](../architecture-guide.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

AddEditSavedAddressPopup: 
Snímek obrazovky z 2024-09-25 17-20-19

DappSignRequestModal:
image

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml [hunk 1]
```diff
@@ -163,15 +163,16 @@ DappsComboBox {
...
+                migratedToKeycard: false
             }
 
             readonly property var network: networkEntry.available ? networkEntry.item : {
                 chainName: "",
                 iconUrl: ""
             }
 
-            loginType: account.migragedToKeycard ? Constants.LoginType.Keycard : root.loginType
+            loginType: account.migratedToKeycard ? Constants.LoginType.Keycard : root.loginType
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml:175]: That typo :laughing:

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 2]
```diff
@@ -180,7 +182,7 @@ StatusModal {
             mainModule.resolveENS(name, d.uuid)
         });
 
-        property var contactsModuleInst: root.store.profileSectionModuleInst.contactsModule
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:183]: Here the `root.store` is not `WalletStores.RootStore` but a `SharedStores.RootStore` in fact

File: ui/app/AppLayouts/Wallet/stores/RootStore.qml [hunk 3]
```diff
@@ -561,6 +561,10 @@ QtObject {
         return transaction.txType
     }
 
+    function addressesEqual(address1, address2) {
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/stores/RootStore.qml:564]: This is referenced in https://github.com/status-im/status-desktop/pull/16406/files#diff-49a543b84eef8685a0154567966dc0bfb8238b733a8beeb4149988a75604f4b8R545 but was missing (forgotten in the component itself)

File: ui/imports/shared/controls/TransactionDelegate.qml [hunk 4]
```diff
@@ -202,10 +202,6 @@ StatusListItem {
         property color animatedBgColor
         property int txType: walletRootStore.transactionType(root.modelData)
 
-        function addressesEqual(address1, address2) {
```
Comment (caybro) [inline ui/imports/shared/controls/TransactionDelegate.qml:205]: Not used here at all but needed in the store now...

**Review by micieslak — APPROVED** [review]
LGTM, Thanks for fixing that!

**Review by saledjenic — APPROVED** [review]

**Review by alexjba — APPROVED** [review]

**Pull request merged**